tool/speed.cc

```diff
@@ -1320,70 +1331,66 @@ bool Speed(const std::vector<std::string> &args) {
-                 kLegacyADLen, selected) ||
-      !SpeedAEADOpen(EVP_aead_aes_128_cbc_sha1_tls(), "AES-128-CBC-SHA1",
-                     kLegacyADLen, selected) ||
-      !SpeedAEADOpen(EVP_aead_aes_256_cbc_sha1_tls(), "AES-256-CBC-SHA1",
-                     kLegacyADLen, selected) ||
-      !SpeedAEAD(EVP_aead_aes_128_gcm_siv(), "AES-128-GCM-SIV", kTLSADLen,
-                 selected) ||
-      !SpeedAEAD(EVP_aead_aes_256_gcm_siv(), "AES-256-GCM-SIV", kTLSADLen,
-                 selected) ||
-      !SpeedAEADOpen(EVP_aead_aes_128_gcm_siv(), "AES-128-GCM-SIV", kTLSADLen,
-                     selected) ||
-      !SpeedAEADOpen(EVP_aead_aes_256_gcm_siv(), "AES-256-GCM-SIV", kTLSADLen,
-                     selected) ||
-      !SpeedAEAD(EVP_aead_aes_128_ccm_bluetooth(), "AES-128-CCM-Bluetooth",
-                 kTLSADLen, selected) ||
-      !SpeedAESBlock("AES-128", 128, selected) ||
-      !SpeedAESBlock("AES-256", 256, selected) ||
-      !SpeedHash(EVP_sha1(), "SHA-1", selected) ||
-      !SpeedHash(EVP_sha256(), "SHA-256", selected) ||
-      !SpeedHash(EVP_sha512(), "SHA-512", selected) ||
-      !SpeedHash(EVP_blake2b256(), "BLAKE2b-256", selected) ||
-      !SpeedRandom(selected) ||
-      !SpeedECDH(selected) ||
-      !SpeedECDSA(selected) ||
-      !Speed25519(selected) ||
-      !SpeedSPAKE2(selected) ||
-      !SpeedScrypt(selected) ||
-      !SpeedRSAKeyGen(selected) ||
-      !SpeedHRSS(selected) ||
-      !SpeedHashToCurve(selected) ||
-      !SpeedTrustToken("TrustToken-Exp1-Batch1", TRUST_TOKEN_experiment_v1(), 1,
-                       selected) ||
-      !SpeedTrustToken("TrustToken-Exp1-Batch10", TRUST_TOKEN_experiment_v1(),
-                       10, selected) ||
-      !SpeedTrustToken("TrustToken-Exp2VOPRF-Batch1",
-                       TRUST_TOKEN_experiment_v2_voprf(), 1, selected) ||
-      !SpeedTrustToken("TrustToken-Exp2VOPRF-Batch10",
-                       TRUST_TOKEN_experiment_v2_voprf(), 10, selected) ||
-      !SpeedTrustToken("TrustToken-Exp2PMB-Batch1",
-                       TRUST_TOKEN_experiment_v2_pmb(), 1, selected) ||
-      !SpeedTrustToken("TrustToken-Exp2PMB-Batch10",
-                       TRUST_TOKEN_experiment_v2_pmb(), 10, selected)) {
+  if(!SpeedAESBlock("AES-128", 128, selected) ||
+     !SpeedAESBlock("AES-192", 192, selected) ||
+     !SpeedAESBlock("AES-256", 256, selected) ||
+     !SpeedHash(EVP_md4(), "MD4", selected) ||
+     !SpeedHash(EVP_md5(), "MD5", selected) ||
+     !SpeedHash(EVP_sha1(), "SHA-1", selected) ||
+     !SpeedHash(EVP_sha224(), "sha-224", selected) ||
+     !SpeedHash(EVP_sha256(), "SHA-256", selected) ||
+     !SpeedHash(EVP_sha384(), "SHA-384", selected) ||
+     !SpeedHash(EVP_sha512(), "SHA-512", selected) ||
+     !SpeedRandom(selected) ||
+     !SpeedECDH(selected) ||
+     !SpeedECDSA(selected) ||
+     !SpeedScrypt(selected) ||
+     !SpeedRSA(selected) ||
+     !SpeedRSAKeyGen(selected)
+#if !defined(OPENSSL_BENCHMARK)
```

Comment:
We miss benchmarks for a number of important algorithms:
* Curve25519-based
* AES-CBC
* AES-GCM

Why not bridge that gap?

Reply:
Unless I'm not understanding something, I think the benchmarks you want are at lines 1367-1368, 1371-1372, and 1380. Or is it that these aren't the benchmarks you're looking for...

Reply:
That's right. Also chacha20-poly1305.

It's not completely straightforward though, because the APIs differ. So, might need some extra benchmark functions that does it in a compatible, unified, and fair way.

Reply:
Let's move ahead without this.